net.go

```diff
@@ -149,3 +149,54 @@ func DialSession(network, addr string, ctx *Ctx, flags DialFlags,
 	}
 	return conn, nil
 }
+
+// DialUpgrade will make the connection conn the underlying connection of an
+// OpenSSL client connection using context ctx.
+// If flags includes InsecureSkipHostVerification, the server certificate's
+// hostname will not be checked to match the hostname in addr. Otherwise, flags
+// should be 0.
+//
+// Dial probably won't work for you unless you set a verify location or add
+// some certs to the certificate store of the client context you're using.
+// This library is not nice enough to use the system certificate store by
+// default for you yet.
+func DialUpgrade(conn net.Conn, addr string, ctx *Ctx,
```

Comment:
Nitpick: I would name this one
```suggestion
func DialUpgrade(tcpConn net.Conn, addr string, ctx *Ctx,
```

and the created SSL connection just `conn`.

This is to be consistent with previous method `DialSession`